Sources/OuterCore/Continuation.swift

```diff
@@ -0,0 +1,130 @@
+/// Continuation.swift
+///
+/// Copyright 2017, The Silt Language Project.
+///
+/// This project is released under the MIT license, a copy of which is
+/// available in the repository.
+
+import Basic
+
+public final class Environment {
+  var uniqueNamePool = 0
+  var counter = [String: Int]()
+
+  func makeUnique(_ string: String?) -> String {
+    guard let string = string else {
+      defer { uniqueNamePool += 1 }
+      return "\(uniqueNamePool)"
+    }
+    let existingCount = counter[string, default: 0]
+    counter[string] = existingCount + 1
+    if existingCount == 0 {
+      return string
+    } else {
+      return makeUnique("\(string).\(existingCount)")
+    }
+  }
+}
+
+public struct ParameterSemantics {
+  var parameter: Parameter
+  var destructor: Destructor?
```

Comment:
This seems like an inversion of priorities to me.  You want to be able to compute a calling convention given the name of a function.  The parameter's calling convention had better be a property of the parameter itself.

Reply:
That's fair. I shouldn't actually put the destructor here and should make this purely a 'needs to be destroyed' bit.